File: project_essai_flutter/lib/mdels/intervention.dart
 
class Intervention {
  final String ?codeIntervention;
    int idIntervention;
  final DateTime? dateHeureIntervention;
  final int? intervenant;
  final bool? cloture;
  final DateTime? dateHeureCloture;
  final String? originePanne;
  final String? remede;
  final String? causePanne;
  final bool? interventionExterne;
  final String? codeDemande;
  final int? idTypeIntervention;
  final int? idPrestataire;
  final bool? changementPiece;
  final bool? interventionPause;
  final int? terminer;
  final String? titre;
  final bool? aideIntervenant;
  final int? createUser;
  final DateTime? createDateTime;
  final int? lastUpdateUser;
  final DateTime? lastUpdateDateTime;
  final String createMachine;
  final DateTime? dateCreateReel;
 

  Intervention({
    required this.codeIntervention,
        required this.idIntervention,
    this.dateHeureIntervention,
    this.intervenant,
    this.cloture,
    this.dateHeureCloture,
    required this.originePanne,
    required this.remede,
    required this.causePanne,
    this.interventionExterne,
    required this.codeDemande,
    this.idTypeIntervention,
    this.idPrestataire,
    this.changementPiece,
    this.interventionPause,
    this.terminer,
    required this.titre,
    this.aideIntervenant,
    this.createUser,
    this.createDateTime,
    this.lastUpdateUser,
    this.lastUpdateDateTime,
    required this.createMachine,
    this.dateCreateReel,
  });

  factory Intervention.fromJson(Map<String, dynamic> json) {
    return Intervention(
      codeIntervention: json['codeIntervention'],
            idIntervention: json['idIntervention'],

      dateHeureIntervention: json['dateHeureIntervention'] != null
          ? DateTime.parse(json['dateHeureIntervention'])
          : null,
      intervenant: json['intervenant'],
      cloture: json['cloture'],
      dateHeureCloture: json['dateHeureCloture'] != null
          ? DateTime.parse(json['dateHeureCloture'])
          : null,
      originePanne: json['originePanne'],
      remede: json['remede'],
      causePanne: json['causePanne'],
      interventionExterne: json['interventionExterne'],
      codeDemande: json['codeDemande'],
      idTypeIntervention: json['idTypeIntervention'],
      idPrestataire: json['idPrestataire'],
      changementPiece: json['changementPiece'],
      interventionPause: json['interventionPause'],
      terminer: json['terminer'],
      titre: json['titre'],
      aideIntervenant: json['aideIntervenant'],
      createUser: json['createUser'],
      createDateTime: json['createDateTime'] != null
          ? DateTime.parse(json['createDateTime'])
          : null,
      lastUpdateUser: json['lastUpdateUser'],
      lastUpdateDateTime: json['lastUpdateDateTime'] != null
          ? DateTime.parse(json['lastUpdateDateTime'])
          : null,
      createMachine: json['createMachine'],
      dateCreateReel: json['dateCreateReel'] != null
          ? DateTime.parse(json['dateCreateReel'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'codeIntervention': codeIntervention,
            'idIntervention': idIntervention,
      'dateHeureIntervention': dateHeureIntervention?.toIso8601String(),
      'intervenant': intervenant,
      'cloture': cloture,
      'dateHeureCloture': dateHeureCloture?.toIso8601String(),
      'originePanne': originePanne,
      'remede': remede,
      'causePanne': causePanne,
      'interventionExterne': interventionExterne,
      'codeDemande': codeDemande,
      'idTypeIntervention': idTypeIntervention,
      'idPrestataire': idPrestataire,
      'changementPiece': changementPiece,
      'interventionPause': interventionPause,
      'terminer': terminer,
      'titre': titre,
      'aideIntervenant': aideIntervenant,
      'createUser': createUser,
      'createDateTime': createDateTime?.toIso8601String(),
      'lastUpdateUser': lastUpdateUser,
      'lastUpdateDateTime': lastUpdateDateTime?.toIso8601String(),
      'createMachine': createMachine,
      'dateCreateReel': dateCreateReel?.toIso8601String(),
    };
  }
}

File: project_essai/project_essai/models/Intervention.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace project_essai.Models
{
    public class Intervention
    {
        [Key]
        public string? CodeIntervention { get; set; }
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? IdIntervention { get; set; }  

        public DateTime? DateHeureIntervention { get; set; }  
        public int? Intervenant { get; set; }  
        public bool? Cloture { get; set; } 
        public DateTime? DateHeureCloture { get; set; } 
        public string? OriginePanne { get; set; }  
        public string? Remede { get; set; } 
        public string? CausePanne { get; set; }  
        public bool? InterventionExterne { get; set; }  
        public string? CodeDemande { get; set; }  
        public int? IdTypeIntervention { get; set; }  
        public int? IdPrestataire { get; set; }  
        public bool? ChangementPiece { get; set; }  
        public bool? InterventionPause { get; set; }  
        public int? Terminer { get; set; } 
        public string? Titre { get; set; } 
        public bool? AideIntervenant { get; set; }  
        public int? CreateUser { get; set; }  
        public DateTime? CreateDateTime { get; set; }  
        public int? LastUpdateUser { get; set; }  
        public DateTime? LastUpdateDateTime { get; set; }  
        public string? CreateMachine { get; set; }  
        public DateTime? DateCreateReel { get; set; }  
    }
}
